api/osrd_infra/views/train_schedule/standalone_simulation.py

```diff
@@ -83,7 +83,40 @@ def run_simulation(request_payload):
     return response.json()
 
 
-def process_simulation_response(infra: Infra, train_schedules: List[TrainScheduleModel], response_payload):
+def _create_and_fill_simulation_output(
+    train_schedule: TrainSchedule,
+    base_simulation: Any,
+    eco_simulation: Any,
+    stops_updates: List[Any],
+    mrsp: Any,
+    modes_and_profiles: Optional[Any],
+) -> SimulationOutput:
+    # Update stops (adding id and name when available)
+    for stop, stop_update in zip(base_simulation["stops"], stops_updates):
+        stop.update(stop_update)
+
+    simulation_output = SimulationOutput(
+        train_schedule=train_schedule,
+        base_simulation=base_simulation,
+        mrsp=mrsp,
+        modes_and_profiles=modes_and_profiles or [],
+    )
+
+    # Skip if no eco simulation is available
+    if not eco_simulation:
+        return simulation_output
+
+    # Update stops (adding id and name when available)
+    for stop, stop_update in zip(eco_simulation["stops"], stops_updates):
+        stop.update(stop_update)
```

Comment:
It seems like this block shouldn't be here (duplicated and weirdly placed), if it should it needs an extra comment